modules/lambdas/main.tf

```diff
@@ -298,6 +298,75 @@ resource "aws_lambda_function" "request_status_for_granule" {
   }
 }
 
+# API Gateway
+resource "aws_api_gateway_rest_api" "request_status_for_granule_api" {
+  name = "${var.prefix}_request_status_for_granule_api"
+}
+
+resource "aws_api_gateway_resource" "request_status_for_granule_api_resource" {
+  path_part   = "{proxy+}"
+  parent_id   = aws_api_gateway_rest_api.request_status_for_granule_api.root_resource_id
+  rest_api_id = aws_api_gateway_rest_api.request_status_for_granule_api.id
+}
+
+resource "aws_api_gateway_method" "request_status_for_granule_api_method" {
+  rest_api_id   = aws_api_gateway_rest_api.request_status_for_granule_api.id
+  resource_id   = aws_api_gateway_resource.request_status_for_granule_api_resource.id
+  http_method   = "ANY"
+  # todo: Make sure this is locked down against external access.
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "request_status_for_granule_api_integration" {
+  rest_api_id             = aws_api_gateway_rest_api.request_status_for_granule_api.id
+  resource_id             = aws_api_gateway_resource.request_status_for_granule_api_resource.id
+  http_method             = aws_api_gateway_method.request_status_for_granule_api_method.http_method
+  integration_http_method = "POST"
+  type                    = "AWS"
+  uri                     = aws_lambda_function.request_status_for_granule.invoke_arn
+}
+
+resource "aws_api_gateway_method_response" "response_200" {
+  rest_api_id = aws_api_gateway_rest_api.request_status_for_granule_api.id
+  resource_id = aws_api_gateway_resource.request_status_for_granule_api_resource.id
+  http_method = aws_api_gateway_method.request_status_for_granule_api_method.http_method
+  status_code = "200"
+}
+
+resource "aws_api_gateway_integration_response" "request_status_for_granule_api_response" {
+  depends_on = [aws_api_gateway_integration.request_status_for_granule_api_integration]
+  rest_api_id = aws_api_gateway_rest_api.request_status_for_granule_api.id
+  resource_id = aws_api_gateway_resource.request_status_for_granule_api_resource.id
+  http_method = aws_api_gateway_method.request_status_for_granule_api_method.http_method
+  status_code = aws_api_gateway_method_response.response_200.status_code
+
+  # Transforms the backend JSON response to XML
+  response_templates = {
+    "application/json" = <<EOF
+#set($inputRoot = $input.path('$'))
+$input.json("$")
+
+#if($input.path("stackTrace") != '')
+#set($context.responseOverride.status = 500)
+#elseif($input.path("httpStatus") != '')
+#set($context.responseOverride.status = $input.path('httpStatus'))
+#end
+EOF
+  }
+}
+
+# Lambda
+resource "aws_lambda_permission" "request_status_for_granule_api_permission" {
```

Comment:
If we do want to lock down to Cumulus' security, the docs I've read suggest we need more here.

Reply:
I feel this is something we definitely want to do long term for security. We will probably want to chat with the Cumulus folks about it to make sure we are all on the same page and that we grab the correct AWS API Gateway ARN that Cumulus will be calling our Gateway from. In the short term we may want to leave this simply locked to the accountID so we could call it without going through Cumulus but, it is not open to the world. I would create a card for security on the API Gateway so we can talk/plan/implement with Cumulus and then setup the security so that the source_arn is not set but the principal is accountId possibly. So at least it is locked to VPC initially. We can pull the accountID through a data structure. See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity. This is just a thought, the main thing is in a Prod environment it would be locked down.

Reply:
Hm, it didn't like that. Said it gave a permission error. We'll stick with this for the time being.

Reply:
https://bugs.earthdata.nasa.gov/browse/ORCA-206